### PR TITLE
Create CRDs for apiVersion v1 and update v1beta1

### PR DESCRIPTION
**Pull request by lucacome**
### Proposed changes
- Generated CRDs v1 with controller-gen v0.4.0 (default to apiextensions.k8s.io/v1)
- Generated CRDs v1beta1 to folder `crds-v1beta1` with controller-gen
- Added directives for generation and fixed some comments/descriptions

### CODE REVIEW

File: docs-web/installation/installation-with-manifests.md [hunk 1]
```diff
@@ -92,8 +94,8 @@ If you would like to use the App Protect module, create the following additional
 1. Create a custom resource definition for `APPolicy` and `APLogConf`:
    
    ```
-   $ kubectl apply -f common/ap-logconf-definition.yaml 
-   $ kubectl apply -f common/ap-policy-definition.yaml 
+   $ kubectl apply -f common/crds-v1beta1/ap-logconf-definition.yaml 
+   $ kubectl apply -f common/crds-v1beta1/ap-policy-definition.yaml
```
Comment (Dean-Coakley) [inline docs-web/installation/installation-with-manifests.md:98]: I think these should also be `/crds`?
```suggestion
   $ kubectl apply -f common/crds/ap-logconf-definition.yaml 
   $ kubectl apply -f common/crds/ap-policy-definition.yaml 
```

**Review by Dean-Coakley — COMMENTED** [review]
@lucacome Are similar changes required for the helm chart? - If not, why not?

File: docs-web/installation/installation-with-manifests.md (hunk 1 above)
Comment (lucacome) [reply docs-web/installation/installation-with-manifests.md:98]: we don't have new CRDs for AP yet

**Comment by lucacome** [comment]
> @lucacome Are similar changes required for the helm chart? - If not, why not?

yeah I think we will need new CRDs for helm chart as well, but that probably would mean dropping support for 1.15

File: docs-web/installation/installation-with-manifests.md [hunk 2]
```diff
@@ -62,18 +62,20 @@ In this section, we create resources common for most of the Ingress Controller i
 ValidationError(CustomResourceDefinition.spec): unknown field "preserveUnknownFields" in io.k8s.apiextensions-apiserver.pkg.apis.api extensions.v1beta1.CustomResourceDefinitionSpec
 ```
 
+**Note**: There are two different sets of Custom Resources, one for Kubernetes <= v1.15 and one for Kubernetes >= v1.16. For Kubernetes <= v1.15 substitute `crds` with `crds-v1beta1` in the following commands.
```
Comment (pleshakov) [inline docs-web/installation/installation-with-manifests.md:65]: do we really need to use crds v1 for Kubernetes >= 1.15? I wonder if can just use v1beta until we stop supporting k8s 1.15. This way we will not need to use different folders in the docs and make any comments 🤔 

we can still store v1 in some folder if those are necessary

Comment (pleshakov) [inline docs-web/installation/installation-with-manifests.md:65]:
```suggestion
**Note**: There are two different sets of custom resource definitions: one for Kubernetes <= v1.15 and one for Kubernetes >= v1.16. For Kubernetes <= v1.15 substitute `crds` with `crds-v1beta1` in the following commands.
```

**Review by pleshakov — COMMENTED** [review]
> yeah I think we will need new CRDs for helm chart as well, but that probably would mean dropping support for 1.15

can we use v1beta1 for now for helm and manifest installations?

File: Makefile [hunk 3]
```diff
@@ -33,14 +33,23 @@ endif
...
-	go run sigs.k8s.io/controller-tools/cmd/controller-gen schemapatch:manifests=./deployments/common/ paths=./pkg/apis/configuration/... output:dir=./deployments/common
+	go run sigs.k8s.io/controller-tools/cmd/controller-gen schemapatch:manifests=./deployments/common/crds/ paths=./pkg/apis/configuration/... output:dir=./deployments/common/crds
+	go run sigs.k8s.io/controller-tools/cmd/controller-gen schemapatch:manifests=./deployments/common/crds-v1beta1/ paths=./pkg/apis/configuration/... output:dir=./deployments/common/crds-v1beta1
+	@make update-crds-helm
+
+update-crds-helm:
+	@for f in deployments/common/crds-v1beta1/*.yaml; do \
+		echo "updating $$(basename $$f) crd for helm"; \
+		kubectl label -f $$f --local=true -o yaml app.kubernetes.io/name=nginx-ingress > deployments/helm-chart/crds/$$(basename $$f); \
+	done
```
Comment (Dean-Coakley) [inline Makefile:52]: Just so you know, these labels are not essential. For the case of simplicity, I think you can remove this Makefile target if you want

```suggestion
```

**Review by Dean-Coakley — APPROVED** [review]
lgtm

One optional suggestion.

**Review by pleshakov — APPROVED** [review]
🚀

**Pull request merged**